src/main/java/org/gridsuite/ds/server/service/DynamicSimulationWorkerService.java

```diff
@@ -112,37 +125,47 @@ private Network getNetwork(UUID networkUuid) {
-
                 run(resultContext.getRunContext())
-                            .flatMap(result -> updateResult(resultContext.getResultUuid(), result.isOk()))
-                            .doOnSuccess(unused -> {
-                                Message<String> sendMessage = MessageBuilder
-                                        .withPayload("")
-                                        .setHeader("resultUuid", resultContext.getResultUuid().toString())
-                                        .build();
-                                sendResultMessage(sendMessage);
-                                LOGGER.info("Dynamic simulation complete (resultUuid='{}')", resultContext.getResultUuid());
-                            }).block();
+                        .flatMap(result -> updateResult(resultContext.getResultUuid(), result))
+                        .doOnSuccess(result -> {
+                            // build json payload
+                            ByteArrayOutputStream bytesOS = new ByteArrayOutputStream();
+                            DynamicSimulationResultSerializer.write(result, bytesOS);
+                            Message<String> sendMessage = MessageBuilder
+                                    .withPayload(bytesOS.toString())
+                                    .setHeader("resultUuid", resultContext.getResultUuid().toString())
+                                    .build();
+                            notificationService.emitResultDynamicSimulationMessage(sendMessage);
+                            LOGGER.info("Dynamic simulation complete (resultUuid='{}')", resultContext.getResultUuid());
+                        })
+                        .toFuture().get();
             } catch (Exception e) {
+                dynamicSimulationWorkerUpdateResult.doUpdateResult(resultContext.getResultUuid(), null, null, DynamicSimulationStatus.NOT_DONE);
                 LOGGER.error("error in consumeRun", e);
+                // S2142 Restore interrupted state...
+                Thread.currentThread().interrupt();
             }
         };
     }
 
-    public Mono<Void> updateResult(UUID resultUuid, Boolean result) {
+    public Mono<DynamicSimulationResult> updateResult(UUID resultUuid, DynamicSimulationResult result) {
         Objects.requireNonNull(resultUuid);
-        return Mono.fromRunnable(() -> dynamicSimulationWorkerUpdateResult.doUpdateResult(resultUuid, result));
-    }
-
-    public void setFileSystem(FileSystem fs) {
-        this.fileSystem = fs;
-    }
-
-    private void sendResultMessage(Message<String> message) {
-        OUTPUT_MESSAGE_LOGGER.debug("Sending message : {}", message);
-        publishResult.send("publishResult-out-0", message);
+        List<TimeSeries> timeSeries = new ArrayList(result.getCurves().values());
+        StringTimeSeries timeLine = result.getTimeLine();
+        return Mono.zip(
+                    timeSeriesClient.sendTimeSeries(timeSeries).subscribeOn(Schedulers.immediate()),
```

Comment:
why SubscribeOn(Schedulers.immediate()) ? It's rare to need this

Reply:
Yes, replaced by boundedElastic we can send in parallel